static/src/javascripts/projects/common/modules/commercial/fiv-banner.js

```diff
@@ -0,0 +1,132 @@
+// @flow
+import config from 'lib/config';
+import { local } from 'lib/storage';
+import { isBlocked } from 'common/modules/commercial/membership-engagement-banner-block';
+import { shouldShowReaderRevenue } from 'common/modules/commercial/contributions-utilities';
+import type { Banner } from 'common/modules/ui/bannerPicker';
+import { acquisitionsBannerFivTemplate } from 'common/modules/commercial/templates/acquisitions-banner-fiv';
+import userPrefs from 'common/modules/user-prefs';
+import {getSync as getGeoLocation} from "../../../../lib/geolocation";
+import {supportContributeURL} from "common/modules/commercial/support-utilities";
+import {addTrackingCodesToUrl, submitComponentEvent} from "common/modules/commercial/acquisitions-ophan";
+import {Message} from "common/modules/ui/message";
+import bean from "bean";
+import mediator from "../../../../lib/mediator";
+
+const messageCode = 'fiv-banner';
+const maxArticles = 3;
+const fivFirstSeenAtKey = 'fivFirstSeenAt';
```

Comment:
this is the gu.alreadyVisited value at the point the fiv was first shown.. then we can know to only show it 3 times.  We can change this to -3 to stop it appearing and distinguish from never seen (although that's a bit cheaty I didn't want to clutter up people's storage with more data)

Reply:
on reflection I could just store a fivImpressionsRemaining and initialise to 3 and decrement it every time, it would involve more mutation but it might be more accurate

Reply:
I think I prefer the second approach as it's clearer, less logic, no dependency on a second `localStorage` entry.

Reply:
I have done this btw, thanks for the suggestion